File: backend/QuizHubApplication/DTOs/Requests/CreateQuestionDto.cs
using QuizHubDomain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizHubApplication.DTOs.Requests
{
    public record CreateQuestionDto(int Order,
        string Text,
        int Points,
        int QuizId,
        QuestionType QuestionType,
        string? CorrectFillInAnswer,
        List<CreateAnswerOptionDto> AnswerOptions
        )
    {
    }
}

File: frontend/QuizHubFrontend/src/app/features/quiz/models/CreateQuestionDto .ts
import { CreateAnswerOptionDto } from './CreateAnswerOptionDto ';

export interface CreateQuestionDto {
    text: string;
    points: number;
    quizId: number;
    questionType: string; // ili enum ako želiš
    correctFillInAnswer?: string;
    answerOptions: CreateAnswerOptionDto[];
}
